FirebaseDatabase/Sources/Core/View/FCacheNode.m

```diff
@@ -30,6 +30,9 @@ @implementation FCacheNode
 - (id)initWithIndexedNode:(FIndexedNode *)indexedNode
        isFullyInitialized:(BOOL)fullyInitialized
                isFiltered:(BOOL)filtered {
+    NSAssert([indexedNode isKindOfClass:[FIndexedNode class]],
+             @"Expected node with type FIndexNode, but instead got %@",
+             indexedNode);
```

Comment:
Will it always lead to a crash if it's not an `FIndexedNode`? Won't `NSAssert` be compiled out in Release mode?

Reply:
Loosened this check to only check for the crashing selector and made it a thrown exception instead of an assert.